### PR TITLE
refactor: defaulter interface

### PR DESCRIPTION
**Pull request by caarlos0**
Right now the code looks weird because the defaults
of a pipe are far away of the implementation of the pipe.

the intend of this PR is to bring them closer by having a
Defaulter interface.

I also renamed the Pipe interface to Piper, and removed
the Description method in favor for fmt.Stringer.

The next step (in another PR) would probably be some sort 
of refactor in the config itself, by bringing each pipe's config
closer to the pipe itself.

What do you guys think?

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] I have read the **CONTRIBUTING** document.
- [ ] `make ci` passes on my machine.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: pipeline/default.go [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package pipeline
+
+import (
+	"fmt"
+
+	"github.com/goreleaser/goreleaser/context"
+)
+
+// Defaulter interface
```
Comment (jorinvo) [inline pipeline/default.go]: Maybe can be a bit more descriptive here, something like:
`Defaulter can be implemented by a Piper to set default values for its configuration.`

File: goreleaserlib/goreleaser.go [hunk 2]
```diff
@@ -26,7 +26,7 @@ import (
 	yaml "gopkg.in/yaml.v2"
 )
 
-var pipes = []pipeline.Pipe{
+var pipes = []pipeline.Piper{
 	defaults.Pipe{},  // load default configs
```
Comment (jorinvo) [inline goreleaserlib/goreleaser.go]: Maybe it would be nicer if `defaults` is not a `Piper` and instead of explicitly listing

Comment (caarlos0) [reply goreleaserlib/goreleaser.go]: the problem is the order of things, the default's order is different from the pipe's order...

Comment (jorinvo) [reply goreleaserlib/goreleaser.go]: Ah missed that part. Never mind then :)

**Pull request merged**